### ISSUE TITLE
Conveyer Behaviour not working in latest build

### ISSUE THREAD

**Issue by synchromesh62** [TheGameCreators/GameGuruRepo#4454]
Conveyer Behaviour not working in latest build,
No errors shown but just not recognised. ?

**Comment by Necrym59** [comment]
Can confirm this, it seems like the "Intersect all" doesnt seem to work anymore with the player capsule when a check is done for the player standing on somthing for example. It used to work perfectly before.
_onconveyor[e]=IntersectAll(g_PlayerPosX,g_PlayerPosY,g_PlayerPosZ,g_Entity[e]['x'],-1,g_Entity[e]['z'],0)_
now reports nothing?? but no errors either.

**Comment by synchromesh62** [comment]
> Can confirm this, it seems like the "Intersect all" doesnt seem to work anymore with the player capsule when a check is done for the player standing on somthing for example. It used to work perfectly before. _onconveyor[e]=IntersectAll(g_PlayerPosX,g_PlayerPosY,g_PlayerPosZ,g_Entity[e]['x'],-1,g_Entity[e]['z'],0)_ now reports nothing?? but no errors either.
> 
> Seems also to impact on the pinpoint collection of items as this uses the IntersectAll command as well.

After further investigation this also seems to break some of your up and coming scripts i have been testing like the weight switch behaviour etc. Im hoping its a fix one fix's all  :)

**Comment by Necrym59** [comment]
Now fixed it thanks guys

**Comment by plemsoft** [comment]
@Necrym59 Not really working here , i changed the ray to go down from the player so it only hit when the objects is below the player :)

	onconveyor[e]=IntersectAll(g_PlayerPosX,g_PlayerPosY,g_PlayerPosZ,g_PlayerPosX,-1,g_PlayerPosZ,0)

**Comment by Necrym59** [comment]
Its working fine on the conveyor now on latest build

**Comment by plemsoft** [comment]
@Necrym59 Its all over the place for me:
Latest EXP:
https://www.youtube.com/watch?v=QRcdqchkkU0

I assume it should only be active when the player is above the conveyor object , thats why i changed it to:

onconveyor[e]=IntersectAll(g_PlayerPosX,g_PlayerPosY,g_PlayerPosZ,g_PlayerPosX,-1,g_PlayerPosZ,0)

**Comment by synchromesh62** [comment]
Confirmed all working now including the others effected i was testing.
Fantastic thanks Guys

**Comment by plemsoft** [comment]
@Necrym59 Strange it do not work here unless i use:

onconveyor[e]=IntersectAll(g_PlayerPosX,g_PlayerPosY,g_PlayerPosZ,g_PlayerPosX,-1,g_PlayerPosZ,0)

I do not see those commits from lee ?

@LeeBamberTGC Did you forgot to commit those changes ? so we did the same thing ? if so i will have to revert this:
https://github.com/TheGameCreators/GameGuruMAX/pull/10/commits/030d472deab7008f7c6f2515b680bafa0f062d21
Let me know :)

**Comment by LeeBamberTGC** [comment]
@plemsoft I have merged our commits and latest now in the main trunk.  Making a new releasecandidate build now.

After I fixed the terrain ray cast bug, as you say the InteractAll command behaved differently, so I added some code to solve it in a different way than yours. Improving the standard InteractAll to detect a terrain hit, then shorten the ray at the point it hit the terrain, then carried to see if an object was struck.  The terrain hit by itself would not result in InteractAll returning true.

I have left your code in that uses the parameter 11 method to also skip terrain testing altogether, giving scripters maximum options.
